src/App.js

```diff
@@ -14,11 +15,143 @@ const goodsFromServer = [
   'Garlic',
 ];
 
-const App = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+const preparedGoods = goodsFromServer.map(good => (
+  {
+    name: good,
+    selected: false,
+    id: uuidv4(),
+  }
+));
+
+class App extends React.Component {
+  state = {
+    goods: preparedGoods.map(good => (good.name === 'Jam' ? {
+      ...good, selected: true,
+    } : { ...good })),
+    selectedGoods: ['Jam'],
+  }
+
+  addGood(good) {
+    this.setState(prevState => ({
+      goods: prevState.goods.map(item => (
+        (item === good) ? {
+          ...item, selected: true,
+        } : { ...item })),
+      selectedGoods: [...prevState.selectedGoods, good.name],
+    }));
+  }
+
+  removeGood(good) {
+    this.setState(prevState => ({
+      goods: prevState.goods.map(item => (
+        (item === good) ? {
+          ...item, selected: false,
+        } : { ...item }
+      )),
+      selectedGoods: prevState.selectedGoods.filter(item => item !== good.name),
+    }));
+  }
+
+  removeSelectedGoods() {
+    this.setState(prevState => ({
+      goods: prevState.goods.map(item => ({
+        ...item,
+        selected: false,
+      })),
+      selectedGoods: [],
+    }));
+  }
+
+  showTitle() {
+    const { selectedGoods } = this.state;
+
+    if (selectedGoods.length === 0) {
+      return 'No goods selected';
+    }
+
+    if (selectedGoods.length === 1) {
+      return `${selectedGoods[0]} is  selected`;
+    }
+
+    return `${selectedGoods.slice(0, -1).join(', ')} 
+        and ${selectedGoods.slice(-1)} are selected`;
+  }
+
+  render() {
+    const { selectedGoods, goods } = this.state;
+
+    return (
+      <div className="App d-flex flex-column align-items-center m-4">
+        <h1 className="text-center">
+          {this.showTitle()}
+        </h1>
+        {selectedGoods.length > 0
+          ? (
+            <p>
+              Number of selected goods:
+              {' '}
+              {selectedGoods.length}
+            </p>
+          )
+          : ''
+        }
+        {selectedGoods.length > 0
+          ? (
+            <button
+              type="button"
+              className="btn btn-danger m-3"
+              onClick={() => {
+                this.removeSelectedGoods();
+              }
```

Comment:
якщо функція removeSelectedGoods не приймає параметрів то стрілочну функцію не потрібно писати 
```suggestion
              onClick={this.removeSelectedGoods}
```